### ISSUE TITLE
Tags/InvalidTypes rejects YARD's semicolon shorthand for fixed-shape Hash types

### ISSUE THREAD

**Issue by lekemula** [mensfeld/yard-lint#171]
## Summary

`Tags/InvalidTypes` rejects the semicolon-separated YARD shorthand for
documenting fixed-shape hashes, e.g.:

```ruby
# @return [Hash{:range => Hash; :severity => Integer; :source, :code, :message => String}]
```

yard-lint reports each component as an invalid type name (`Hash;`, `Integer;`,
`String;`).

## Why this is valid YARD

YARD itself accepts this form — see lsegal/yard#1630, which introduced support
for grouping multiple `key => value` pairs in a single `Hash{...}` type using
`;` as the separator (and `,` to share a value type across several keys).

`Tags/TypeSyntax` already accepts the syntax via YARD's own parser; only
`Tags/InvalidTypes` flags it (it appears to split on `;` and treat each fragment
as a standalone type name).

## Reproduction

```ruby
# @return [Hash{:a => Integer; :b => String}]
def foo; end
```

Run `yard-lint`:

```
InvalidTagType: ... has invalid type(s): @return: `Integer;`, `String}`
```

## Note

This case looks like it slipped through #157 — that PR likely focused on the
single-pair `Hash{K => V}` form and didn't cover the multi-pair semicolon
variant.

## Workaround

Disable `Tags/InvalidTypes` in `.yard-lint.yml`.

**Comment by mensfeld** [comment]
Thank you. Fix has been merged. I will do my best to craft a release this week.
